flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/KafkaConsumerThread.java

```diff
@@ -219,6 +209,15 @@ public void run() {
 					}
 				}
 
+				try {
+					newPartitions = unassignedPartitionsQueue.pollBatch();
+					if (newPartitions != null) {
+						reassignPartitions(newPartitions);
+					}
+				} catch (AbortedReassignmentException e) {
+					continue;
```

Comment:
the reason is that the `AbortedReassignmentException` is thrown only if the Kafka consumer was `wakeup` before the reassignment started and that wakeup effect was consumed by an operation during the reassignment.

Now, if the loop just continues, the next Kafka operation (the `poll`) would not be woken up because the wakeup was already consumed.
So, effectively, it would mean the same thing to do this in the catch block instead of falling through the loop:
```
} catch (AbortedReassignmentException e) {
consumer.wakeup();
}
```

I'm just eagerly continuing instead of restoring the wakeup. Does that make sense?